src/darwin/Framework/CHIP/CHIPDevice.h

```diff
@@ -175,9 +175,9 @@ extern NSString * const kCHIPArrayValueType;
 /**
  * Read attribute in a designated attribute path
  */
-- (void)readAttributeWithEndpointId:(NSNumber * _Nullable)endpointId
-                          clusterId:(NSNumber * _Nullable)clusterId
-                        attributeId:(NSNumber * _Nullable)attributeId
+- (void)readAttributeWithEndpointId:(id _Nullable)endpointId
+                          clusterId:(id _Nullable)clusterId
+                        attributeId:(id _Nullable)attributeId
```

Comment:
So now that I am looking at this API... this is really pretty annoying to use: you have to create three lists that line up, etc.

I think we should have a public `readAttributePaths` API, alongside the existing `readAttributeWithEndpointId`, which can call `readAttributePaths` internally.  And readAttributePaths should take a single list of paths, where each path is an object with three nullable members for endpoint/cluster/attribute, null representing wildcard.